formula/src/main/java/com/instacart/formula/internal/ActionDelegate.kt

```diff
@@ -0,0 +1,27 @@
+package com.instacart.formula.internal
+
+import com.instacart.formula.DeferredAction
+import com.instacart.formula.plugin.FormulaError
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * Interface required to start and stop [DeferredAction].
+ */
+internal interface ActionDelegate {
```

Comment:
Instead of passing `FormulaManagerImpl`, we pass `ActionDelegate` to `DeferredAction`